crates/celo-revm/src/api/exec.rs

```diff
@@ -1,64 +1,54 @@
-        Cfg, ContextTr, Database, JournalTr,
+        ContextTr, Database,
         result::{EVMError, ExecutionResult, ResultAndState},
     },
-    handler::{
-        EthFrame, EvmTr, Handler, PrecompileProvider, SystemCallTx, instructions::EthInstructions,
-        system_call::SystemCallEvm,
-    },
-    inspector::{InspectCommitEvm, InspectEvm, Inspector, InspectorHandler, JournalExt},
-    interpreter::{InterpreterResult, interpreter::EthInterpreter},
+    handler::{EthFrame, EvmTr, Handler, SystemCallTx, system_call::SystemCallEvm},
+    inspector::{InspectCommitEvm, InspectEvm, Inspector, InspectorHandler},
+    interpreter::interpreter::EthInterpreter,
 };
-
-// Type alias for Celo context
-pub trait CeloContextTr:
-    ContextTr<
-        Journal: JournalTr<FinalOutput = JournalOutput>,
-        Tx: CeloTxTr,
-        Cfg: Cfg<Spec = OpSpecId>,
-        Chain = CeloBlockEnv,
-    >
-{
-}
-
-impl<T> CeloContextTr for T where
-    T: ContextTr<
-            Journal: JournalTr<FinalOutput = JournalOutput>,
-            Tx: CeloTxTr,
-            Cfg: Cfg<Spec = OpSpecId>,
-            Chain = CeloBlockEnv,
-        >
-{
-}
+use revm_context::ContextSetters;
 
 /// Type alias for the error type of the CeloEvm.
 // TODO: replace with CeloTransactionError
 type CeloError<CTX> = EVMError<<<CTX as ContextTr>::Db as Database>::Error, OpTransactionError>;
 
-impl<CTX, INSP, PRECOMPILE> ExecuteEvm
-    for CeloEvm<CTX, INSP, EthInstructions<EthInterpreter, CTX>, PRECOMPILE>
+impl<DB, INSP> ExecuteEvm for CeloEvm<DB, INSP>
 where
-    CTX: CeloContextTr + ContextSetters,
-    PRECOMPILE: PrecompileProvider<CTX, Output = InterpreterResult>,
+    DB: Database,
 {
-    type Output = Result<ResultAndState<OpHaltReason>, CeloError<CTX>>;
+    type Output = Result<ResultAndState<OpHaltReason>, CeloError<CeloContext<DB>>>;
 
-    type Tx = <CTX as ContextTr>::Tx;
+    type Tx = <CeloContext<DB> as ContextTr>::Tx;
 
-    type Block = <CTX as ContextTr>::Block;
+    type Block = <CeloContext<DB> as ContextTr>::Block;
 
     fn set_tx(&mut self, tx: Self::Tx) {
         self.0.ctx().set_tx(tx);
     }
 
     fn set_block(&mut self, block: Self::Block) {
         self.0.ctx().set_block(block);
+
+        // Update the chain environment with fee currencies
+        // Warning: If ctx.set_block is called directly, the fee currencies are not updated. I
+        // would prefer to place this code in that function, but that would require forking the
+        // Context struct.
+        match CeloBlockEnv::update_fee_currencies(self) {
+            Ok(updated_block_env) => {
+                *self.0.ctx().chain() = updated_block_env;
+            }
+            Err(_e) => {
+                // Log and continue. It's better to continue with outdated or unset fee currency
+                // information rather than shutting down the node.
+                #[cfg(feature = "std")]
+                eprintln!("Failed to update fee currencies in set_block: {:?}", _e);
```

Comment:
The previous one is only working on std environment.
```suggestion
                tracing::error!("Failed to update fee currencies in set_block: {:?}", e);
```